x-pack/plugins/cases/public/containers/use_get_cases.tsx

```diff
@@ -160,39 +167,41 @@ export const useGetCases = (
     dispatch({ type: 'UPDATE_FILTER_OPTIONS', payload: newFilters });
   }, []);
 
-  const fetchCases = useCallback(async (filterOptions: FilterOptions, queryParams: QueryParams) => {
-    try {
-      didCancelFetchCases.current = false;
-      abortCtrlFetchCases.current.abort();
-      abortCtrlFetchCases.current = new AbortController();
-      dispatch({ type: 'FETCH_INIT', payload: 'cases' });
-
-      const response = await getCases({
-        filterOptions,
-        queryParams,
-        signal: abortCtrlFetchCases.current.signal,
-      });
-
-      if (!didCancelFetchCases.current) {
-        dispatch({
-          type: 'FETCH_CASES_SUCCESS',
-          payload: response,
+  const fetchCases = useCallback(
+    async (filterOptions: FilterOptions, queryParams: QueryParams) => {
+      try {
+        didCancelFetchCases.current = false;
+        abortCtrlFetchCases.current.abort();
+        abortCtrlFetchCases.current = new AbortController();
+        dispatch({ type: 'FETCH_INIT', payload: 'cases' });
+
+        const response = await getCases({
+          filterOptions,
+          queryParams,
+          signal: abortCtrlFetchCases.current.signal,
         });
-      }
-    } catch (error) {
-      if (!didCancelFetchCases.current) {
-        if (error.name !== 'AbortError') {
-          errorToToaster({
-            title: i18n.ERROR_TITLE,
-            error: error.body && error.body.message ? new Error(error.body.message) : error,
-            dispatchToaster,
+
+        if (!didCancelFetchCases.current) {
+          dispatch({
+            type: 'FETCH_CASES_SUCCESS',
+            payload: response,
           });
         }
-        dispatch({ type: 'FETCH_FAILURE', payload: 'cases' });
+      } catch (error) {
+        if (!didCancelFetchCases.current) {
+          if (error.name !== 'AbortError') {
+            errorToToaster({
+              title: i18n.ERROR_TITLE,
+              error: error.body && error.body.message ? new Error(error.body.message) : error,
+              dispatchToaster,
+            });
+          }
+          dispatch({ type: 'FETCH_FAILURE', payload: 'cases' });
+        }
       }
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+    },
+    [dispatchToaster]
```

Comment:
I remember this causing re-renders but I am not 100% sure.

Reply:
ok ill remove it